File: src/bin/infer_types.cpp
#include <iostream>

void case1() {
    std::cout << "Case 1:" << std::endl;
    auto a = 1; // int
    long long b = 2;
    std::cout << "Is `inferred` the same type as `typed`? " << (typeid(a) == typeid(b)) << std::endl;
    auto _ = a + b; // long long
}
void case2() {
    std::cout << "Case 2:" << std::endl;
    auto a = 1; // int
    long long b = 2;
    std::cout << "Is `inferred` the same type as `typed`? " << (typeid(a) == typeid(b)) << std::endl;
}

int main() {
    case1();
    case2();
    return 0;
}

File: src/bin/infer_types.rs
use std::any::Any;

fn case1() {
    println!("Case 1:");
    let inferred = 0;
    let typed: i64 = 0;
    println!(
        "Is `inferred` the same type as `typed`? {}",
        inferred.type_id() == typed.type_id()
    );
    let _ = inferred + typed;
}
fn case2() {
    println!("Case 2:");
    let inferred = 0;
    let typed: i64 = 0;
    println!(
        "Is `inferred` the same type as `typed`? {}",
        inferred.type_id() == typed.type_id()
    );
}

fn main() {
    case1();
    case2();
}
